pandas/core/arrays/datetimelike.py

```diff
@@ -1264,13 +1264,18 @@ def min(self, axis=None, skipna=True, *args, **kwargs):
         Series.min : Return the minimum value in a Series.
         """
         nv.validate_min(args, kwargs)
-        nv.validate_minmax_axis(axis)
+        nv.validate_minmax_axis(axis, self.ndim)
 
-        result = nanops.nanmin(self.asi8, skipna=skipna, mask=self.isna())
-        if isna(result):
-            # Period._from_ordinal does not handle np.nan gracefully
-            return NaT
-        return self._box_func(result)
+        if is_period_dtype(self.dtype):
```

Comment:
hmm, why are you converting to 'S' as opposed to something else? can you comment

Reply:
actually found a more performant way to do this, will update